src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationServiceRunner.cs

```diff
@@ -123,13 +121,10 @@ private ImmutableArray<ITypeSymbol> GetTypeSymbols(ImmutableArray<ISymbol> candi
                             type = allTypeArguments[0];
                         }
 
-                        // If type is <see cref="Func{TSomeArguments, TResult}"/>, extract TSomeArguments 
-                        // and find the type of thee argument corresponding to the ordinal.
-                        if (type is INamedTypeSymbol funcSymbolNamedTypeCandidate &&
-                            Equals(funcSymbolNamedTypeCandidate.Name, funcSymbol.Name) &&
-                            Equals(funcSymbolNamedTypeCandidate.ContainingNamespace, funcSymbol.ContainingNamespace))
+                        if (type.IsDelegateType())
                         {
                             var allTypeArguments = type.GetAllTypeArguments();
```

Comment:
What if the delegate type isn't one with type arguments? Doesn't this have to go looking to the Invoke method or something to get the types of the arguments that way?

Reply:
@jasonmalinowski ,  I have added tests for non-generics. Two non-generics work fine (VB and C#). However, one non-generic and another generic do not work together. Semantic models return just a single candidate in those cases. And those candidate even differ in C# and VB. See https://github.com/dotnet/roslyn/issues/35100 for details.

Please let me know if you mean some other tests.

Reply:
My concern isn't the methods being generic, but non-generic delegate types, like:

```
    public delegate void Delegate1(Uri u);
    public delegate void Delegate2(Guid g);

    class Program
    {
        public void M(Delegate1 d) { }
        public void M(Delegate2 d) { }

        public void Test()
        {
            M(d => d.)
        }
    }
```

In this case you should be merging the intellisense for Uri and Guid, correct?